frontend/ui/widget/notification.lua

```diff
@@ -48,44 +53,73 @@ function Notification:init()
         end
     end
 
-    -- we construct the actual content here because self.text is only available now
     local text_widget = TextWidget:new{
         text = self.text,
         face = self.face,
     }
     local widget_size = text_widget:getSize()
-    self[1] = CenterContainer:new{
-        dimen = Geom:new{
-            w = Screen:getWidth(),
-            h = math.floor(Screen:getHeight() / 10),
-        },
-        FrameContainer:new{
-            background = Blitbuffer.COLOR_WHITE,
-            radius = 0,
-            margin = self.margin,
-            padding = self.padding,
-            CenterContainer:new{
-                dimen = Geom:new{
-                    w = widget_size.w,
-                    h = widget_size.h
-                },
-                text_widget,
-            }
+    self.frame = FrameContainer:new{
+        background = Blitbuffer.COLOR_WHITE,
+        radius = 0,
+        margin = self.margin,
+        padding = self.padding,
+        CenterContainer:new{
+            dimen = Geom:new{
+                w = widget_size.w,
+                h = widget_size.h
+            },
+            text_widget,
         }
     }
+    local notif_height = self.frame:getSize().h
+
+    self:_cleanShownStack()
+    table.insert(Notification._nums_shown, os.time())
+    self.num = #Notification._nums_shown
+
+    self[1] = VerticalGroup:new{
+        align = "center",
+        -- We use Spans to properly position this notification:
+        -- have this VerticalGroup full width, to ensure centering
+        HorizontalSpan:new{ width = Screen:getWidth() },
+        -- push this frame at its y=self.num position
+        VerticalSpan:new{ width = notif_height * (self.num - 1) },
```

Comment:
`width`? Not familiar with VerticalSpan, but given the comment, I would have expected `height`?

Reply:
Okay, looking at VerticalSpân, that' sjust verticalSPan itself being *super weird*:

```lua
local VerticalSpan = Widget:new{
    width = 0,
}

function VerticalSpan:getSize()
    return {w = 0, h = self.width}
end

return VerticalSpan
```

o_O

Reply:
Yes, they both have this paramenter named `width`.

I just discovered we have a frontend/ui/widget/rectspan.lua that accepts width + height and that I could use instead of these 2 Horiz+Vert. This RectSpan is currently not used by anything !